rosetta/management/commands/compilemessages.py

```diff
@@ -0,0 +1,100 @@
+from __future__ import unicode_literals
+
+import glob
+import os
+
+from django.core.management.base import CommandError
+from django.core.management.utils import find_command
+from django.utils._os import npath, upath
+from django.core.management.commands.compilemessages import Command as OriginCommand
+from django.core.management.commands.compilemessages import has_bom, is_writable
+
+from rosetta.angular_translation_storage import get_translations_storage
+from rosetta.conf import settings as rosetta_settings
+from rosetta.utils import convert
+
+
+class Command(OriginCommand):
+    def handle(self, **options):
+        locale = options['locale']
+        exclude = options['exclude']
+        self.verbosity = options['verbosity']
+        if options['fuzzy']:
+            self.program_options = self.program_options + ['-f']
+
+        if find_command(self.program) is None:
+            raise CommandError("Can't find %s. Make sure you have GNU gettext "
+                               "tools 0.15 or newer installed." % self.program)
+
+        basedirs = [os.path.join('conf', 'locale'), 'locale']
+        if os.environ.get('DJANGO_SETTINGS_MODULE'):
+            from django.conf import settings
+            basedirs.extend(upath(path) for path in settings.LOCALE_PATHS)
+
+        # Walk entire tree, looking for locale directories
+        for dirpath, dirnames, filenames in os.walk('.', topdown=True):
+            for dirname in dirnames:
+                if dirname == 'locale':
+                    basedirs.append(os.path.join(dirpath, dirname))
+
+        # Gather existing directories.
+        basedirs = set(map(os.path.abspath, filter(os.path.isdir, basedirs)))
+
+        if not basedirs:
+            raise CommandError("This script should be run from the Django Git "
+                               "checkout or your project or app tree, or with "
+                               "the settings module specified.")
+
+        # Build locale list
+        all_locales = []
+        for basedir in basedirs:
+            locale_dirs = filter(os.path.isdir, glob.glob('%s/*' % basedir))
+            all_locales.extend(map(os.path.basename, locale_dirs))
+
+        # Account for excluded locales
+        locales = locale or all_locales
+        locales = set(locales) - set(exclude)
+
+        for basedir in basedirs:
+            if not locales:
```

Comment:
What is point of this check inside cycle? it seems is not populated inside cycle.